test_framework/tacky/common.py

```diff
@@ -220,3 +220,17 @@ def is_zero_instr(i: asm.AsmItem) -> bool:
         and i.opcode == Opcode.XOR
         and i.operands[0] == i.operands[1]
     )
+
+def check_constant_folds(parsed_asm, program_source_file):
```

Comment:
This refactor isn't quite right and fixing it involves a few different changes. Would you rather fix it or just remove it from this PR and save it for a follow-up?

Reply:
Thanks for pointing out the issues. I can go ahead and fix them within this PR.

Reply:
Sounds good, thank you!